apps/webapp-source/setup.sh

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env bash
+
+######
+# setup.sh
+#
+# Sets up infrastructure for a continuously-built and deployed container-based
+# web app for Azure App Service.
+#
+# $1: repo_name: Name of GitHub repo, e.g. joshgav/go-sample.
+# $2: base_name: A default prefix for all Azure resources. Defaults to $AZURE_BASE_NAME.
+# $3: app_name: Name of App Service web app to use or create. Defaults to
+#     "{AZURE_BASE_NAME}-webapp".
+# $4: plan_name: Name of App Service plan to use or create. Defaults to
+#     "{AZURE_BASE_NAME}-plan".
+# $5: group_name: Name of Azure resource group to use for all resources.
+#     Defaults to "${AZURE_BASE_NAME-group".
+# $6: location: Name of Azure location to use for all resources. Defaults to
+#     $AZURE_DEFAULT_LOCATION.
+# $7: gh_token: A GitHub personal access token to set up continuous
+#     integration. Defaults to $GH_TOKEN.
+#####
+
+__filename=${BASH_SOURCE[0]}
+__dirname=$(cd $(dirname ${__filename}) && pwd)
+__root=${__dirname}
+if [[ ! -f "${__root}/.env" ]]; then cp "${__root}/.env.tpl" "${__root}/.env"; fi
+if [[ -f "${__root}/.env" ]]; then source "${__root}/.env"; fi
+source "${__dirname}/scripts/rm_helpers.sh"  # for ensure_group
+
+repo_name=${1:-${REPO_NAME}}
+base_name=${2:-"${AZURE_BASE_NAME}"}
+app_name=${3:-"${base_name}-webapp"}
+plan_name=${4:-"${base_name}-plan"}
+group_name=${5:-"${base_name}-group"}
+location=${6:-${AZURE_DEFAULT_LOCATION}}
+gh_token=${7:-${GH_TOKEN}}
+
+scratch_runtime_id='node|8.1'
+go_runtime_id='go|1'
+url_suffix=azurewebsites.net
+# as specified here:
+# <https://github.com/projectkudu/KuduScript/blob/master/lib/templates/deploy.bash.go.template#L8>
+binary_name=app
+
+# errors
+declare -i err_nogithubtoken=102
+if [[ -z ${gh_token} ]]; then
+    echo 'specify a GitHub personal access token in the env var `GH_TOKEN`' \
+         'to set up continuous integration'
+    exit $err_nogithubtoken
+fi
+
+####
+
+## ensure groups
+ensure_group $group_name
+
+## ensure App Service plan
+plan_id=$(az appservice plan show \
+    --name ${plan_name} \
+    --resource-group ${group_name} \
+    --output tsv --query id)
+
+if [[ -z $plan_id ]]; then
+    plan_id=$(az appservice plan create \
```

Comment:
If I remember right, this creates a plan with a basic `S1` tier (`FREE` is not allowed for Linux containers for some reason) so users should be advised to clean up when finished. Maybe a cleanup script should be provided as well?